src/mechanics/tools/EquationSystem.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include "mechanics/cell/CellInterface.h"
+#include "mechanics/cell/SimpleAssembler.h"
+#include "base/Group.h"
+#include "mechanics/dofs/GlobalDofVector.h"
+#include "mechanics/dofs/GlobalDofMatrixSparse.h"
+
+#include "mechanics/tools/NodalValueMerger.h"
+
+namespace NuTo
+{
+
+//! Equation system that contains Gradient(u) and its derivative with respect to u
+class EquationSystem
+{
+public:
+    EquationSystem(MeshFem* rMesh);
+
+    void SetDofInfo(DofInfo dofInfo);
+
+    void AddGradientFunction(Group<CellInterface> group, CellInterface::VectorFunction f);
+    void AddHessian0Function(Group<CellInterface> group, CellInterface::MatrixFunction f);
+    void AddUpdateFunction(Group<CellInterface> group, CellInterface::VoidFunction f);
+
+    GlobalDofVector Gradient(GlobalDofVector dofValues, std::vector<DofType> dofs);
+    GlobalDofMatrixSparse Hessian0(GlobalDofVector dofValues, std::vector<DofType> dofs);
```

Comment:
I realized that our mechanics integrands need a `dt`. I can imagine some time dependent neumann boundary integrand that uses `t`. So we have to be very careful when binding these functions.

Reply:
The Static problem should store the last update state (history data is already stored, but the corresponding dof_state as well as the time should be a member variable). Thus, the Hessian and well as the gradient function and update function should IMO include t. dt is then automatically computed in the problem and passed to the integrands. I also still think that the interface for the gradienthessian functions for the integrands that we use in a Newmark scheme should be fixed (i.e. gradient is a function of d,v,a,t,dt and cell/ipdata). In this static case, it would be d,t,dt,cell/ip data). Why is the dof type (dofs) a parameter? Wouldn't it make sense to store this as a member variable that is initialized when creating the problem?